ChatSecure/Classes/Controllers/OTRProtocolManager.m

```diff
@@ -206,17 +207,37 @@ - (void)goAwayForAllAccounts {
     }
 }
 
-- (void)disconnectAllAccountsSocketOnly:(BOOL)socketOnly {
+- (void)disconnectAllAccountsSocketOnly:(BOOL)socketOnly timeout:(NSTimeInterval)timeout completionBlock:(nullable void (^)())completionBlock
+{
     @synchronized (self) {
-        [self.protocolManagers enumerateKeysAndObjectsUsingBlock:^(id key, id <OTRProtocol> protocol, BOOL *stop) {
-            [protocol disconnectSocketOnly:socketOnly];
-        }];
+        dispatch_group_t group = dispatch_group_create();
+        for (NSObject<OTRProtocol> *manager in self.protocolManagers.allValues) {
+            if (manager.connectionStatus != OTRProtocolConnectionStatusDisconnected) {
+                dispatch_group_enter(group);
+                __block NSString *token = nil;
+                token = [manager addObserverForKeyPath:@"connectionStatus"
```

Comment:
Instead of using a string literal it's slightly better to use `NSStringFromSelector(@selector(connectionStatus))`